### PR TITLE
Allows a PricingPromise to be provided to the ApplePay constructor

### PR DESCRIPTION
**Pull request by chrissrogers**
- Coerces the underlying pricing instance from the PricingPromise

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 const {
-  BROWSER,
+  BROWSER = 'Chrome',
```
Comment (chrissrogers) [inline karma.conf.js:2]: This just defaults unit test runs to Chrome

File: types/lib/apple-pay.d.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ export type ApplePayConfig = {
    * If provided, will override `options.total` and provide the current total price on the CheckoutPricing instance
    * when the Apple Pay flow is initiated.
    */
-  pricing?: CheckoutPricingInstance;
+  pricing?: CheckoutPricingInstance | CheckoutPricingPromise;
```
Comment (chrissrogers) [inline types/lib/apple-pay.d.ts:29]: This method technically will accept a generic 'PricingInstance' or 'PricingPromise', but our intent is to focus on `CheckoutPricing` going forward.

File: lib/recurly/apple-pay.js [hunk 3]
```diff
@@ -191,6 +186,36 @@ class ApplePay extends Emitter {
     });
   }
 
+  /**
+   * Configures the methodology for pricing
+   *
+   * If a Pricing or PricingPromise instance is available, it will be used. Otherwise, a total
+   * will be expected.
+   *
+   * @param  {Object} options
```
Comment (csampson) [inline lib/recurly/apple-pay.js:195]: Would probably be good to document the properties here

File: lib/recurly/apple-pay.js [hunk 4]
```diff
@@ -191,6 +186,36 @@ class ApplePay extends Emitter {
...
+    } else if ('total' in options) {
+      this.config.total = options.total;
+    } else {
+      throw this.error('apple-pay-config-missing', { opt: 'total' });
+    }
+
+    if (pricing) {
+      this.config.pricing = pricing;
+      pricing.on('change', () => this.onPricingChange());
+      if (pricing.hasPrice) this.onPricingChange();
```
Comment (csampson) [inline lib/recurly/apple-pay.js:215]: Could use a short comment explaining what's going on here

**Review by csampson — CHANGES_REQUESTED** [review]

File: lib/recurly/apple-pay.js (hunk 3 above)
Comment (chrissrogers) [reply lib/recurly/apple-pay.js:195]: They’re documented in the constructor, so just didn’t want to repeat myself here.

**Review by csampson — APPROVED** [review]

**Pull request merged**